File: 11_November_Batch/C/binarySearch.c
#include<stdio.h>
int binarySearch(int n, int arr[],int x){
    int st=0;
    int end=n-1;
    int mid =(st+end)/2;
    while(st<=end){
    if (arr[mid]==x){
        return mid;
    }
    else if(arr[mid]<x){
        st=mid+1;
        mid =(st+end)/2;

    }
    else{
        end=mid-1;
        mid =(st+end)/2;
    }
    }
    return -1;
}

int main(){
    int a,x;
    scanf("%d%d",&a,&x);
    int arr[a];
    

    return 0;
}

File: 11_November_Batch/Python/Array/binarySearch.py
def binarySearch(n,arr,x):
    start=0
    end=n-1
    mid=(start+end)//2
    while(start<=end):
        if (arr[mid]==x):
            return True
        elif (x>arr[mid]):
            start=mid+1
            mid=(start+end)//2
        else:
            end=mid-1
            mid=(start+end)//2
    return False

n,x=map(int,input().split())
arr=list(map(int,input().split()))
ans=binarySearch(n,arr,x)
print(ans)